framework/status/__init__.py

```diff
@@ -56,23 +59,29 @@ def push_status_message(message, kind='warning', dismissible=True, trust=True, j
                            id=id,
                            extra=extra,
                            trust=trust))
-    session.data['status'] = statuses
-    session.save()
+    current_session['status'] = statuses
+    current_session.save()
+
 
 def pop_status_messages(level=0):
-    messages = session.data.get('status')
+    current_session = get_session()
+    session_data = current_session.get_decoded()
+    messages = session_data.get('status', None)
     for message in messages or []:
         if len(message) == 5:
             message += [None, None]  # Make sure all status's have enough arguments
-    session.status_prev = messages
-    if 'status' in session.data:
-        del session.data['status']
-        session.save()
+    current_session['status_prev'] = messages
+    if messages:
+        current_session['status'] = None
```

Comment:
Is there a different between setting a value to a key to `del current_session['status']` and `current_session['status'] = None`? I think the former deletes the key from the dictionary while the latter keeps the key in the dictionary and sets the value to `None`. I don't know whether the difference matter though.

Reply:
Good question! There is no functional difference but I was not sure of the `del`. But looks like it should work: https://docs.djangoproject.com/en/4.1/topics/http/sessions/#django.contrib.sessions.backends.base.SessionBase.__delitem__ In addition, we also need to use `in` check.

Maybe a even better solution is to call `pop`: https://docs.djangoproject.com/en/4.1/topics/http/sessions/#django.contrib.sessions.backends.base.SessionBase.pop